src/storage/src/client.rs

```diff
@@ -574,6 +574,24 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn test_read_object_error_credentials() -> Result {
+        let client = Storage::builder()
+            .with_endpoint("http://private.googleapis.com")
+            .with_credentials(auth::credentials::testing::error_credentials(false))
+            .build()
+            .await?;
+
+        client
+            .read_object()
+            .set_bucket("projects/_/buckets/bucket")
+            .set_object("object")
+            .http_request_builder()
+            .await
+            .expect_err("invalid credentials should err");
```

Comment:
I think we should confirm the type of error

https://github.com/googleapis/google-cloud-rust/blob/827a0b3f336edeb1cb5bf2adf94a9abeacf0c484/src/gax-internal/tests/http_auth.rs#L149-L154

Reply:
Done. Used is_authentication() and inspect_err